Client/Frontend/Browser/ContextMenuHelper.swift

```diff
@@ -84,17 +86,18 @@ class ContextMenuHelper: NSObject, TabHelper, UIGestureRecognizerDelegate {
         return true
     }
 
+    // Hack to detect the built-in context menu gesture recognizer.
     func gestureRecognizer(_ gestureRecognizer: UIGestureRecognizer, shouldBeRequiredToFailBy otherGestureRecognizer: UIGestureRecognizer) -> Bool {
-        // Hack to detect the built-in text selection gesture recognizer.
-        if let otherDelegate = otherGestureRecognizer.delegate, String(describing: otherDelegate).contains("_UIKeyboardBasedNonEditableTextSelectionGestureController") {
+        // On iOS 11 the gestureRecognizer has been renamed. Check for both names.
+        let gestureNames = ["_UIKeyboardBasedNonEditableTextSelectionGestureCluster", "_UIKeyboardBasedNonEditableTextSelectionGestureController"]
+        if let otherDelegate = otherGestureRecognizer.delegate, gestureNames.reduce(false, { $0 || String(describing: otherDelegate).contains($1) }) {
             selectionGestureRecognizer = otherGestureRecognizer
         }
-
-        // Hack to detect the built-in context menu gesture recognizer.
-        return otherGestureRecognizer is UILongPressGestureRecognizer && otherGestureRecognizer.delegate?.description.range(of: "WKContentView") != nil
```

Comment:
Don't we still need to check for `otherGestureRecognizer is UILongPressGestureRecognizer`?

Reply:
There isnt always a longpressGesture on newer devices there is a ForceTouchGesture instead (or both). 

I dont think its required. Instead we decide if we would like to fail our gesture in `gestureRecognizerShouldBegin` which works well.